views/index.slim

```diff
@@ -1,13 +1,31 @@
-table
-  thead
-    th From
-    th Departure time
-    th To
-    th Arrival time
-  tbody
-  - @routes.each do |route|
-    tr
-      td =route[:from]
-      td =route[:departure_datetime]
-      td =route[:to]
-      td =route[:arrival_datetime]
+.search-form-wrapper
+  form action='/' method='post'
+    label From:
+    input#start
+    input#start-id type="hidden" name='departure_id'
+    label To:
+    input#destination
+    input#destination-id type="hidden" name='arrival_id'
+    input type='submit' value='Search'
+
+.table-wrapper
+  table
+    thead
+      tr
+        th Departure station
+        th Departure time
+        th Arrival station
+        th Arrival time
+    tbody
+      - if @routes && !@routes.empty?
+        - @routes.each do |route|
+          tr
+            td=route[:departure_station]
+            td=route[:departure_datetime]
+            td=route[:arrival_station]
+            td=route[:arrival_datetime]
+      - else
+        tr
+          td.empty-table colspan='4' Nothing to show
+
+script src='/js/dashboard.js'
```

Comment:
http://brandonhilkert.com/blog/page-specific-javascript-in-rails/
Same can be applied here